src/config.rs

```diff
@@ -0,0 +1,233 @@
+            Err(_) => panic!("Could not load configuration file contents!"),
+            Ok(s) => s
+        };
+
+        {
+            let (compatible, version) = check_version(&config_content);
+            if ! compatible {
+                eprintln!("The config file version ({}) is not compatible with your program version ({}.{})!\nPlease inspect the Changelog (https://github.com/MaFeLP/discord_bots/releases) and see how to change the config file accordingly!",
+                          version,
+                          env!("CARGO_PKG_VERSION_MAJOR"),
+                          env!("CARGO_PKG_VERSION_MINOR")
+                );
+                exit(1);
+            }
+        }
+
+        let out: Config = match toml::from_str(&config_content) {
+            Err(_) => {
+                let example_config_file = format!("{}.example", config_file);
+                make_default_config(&example_config_file);
+                eprintln!("Configuration file invalid!\nAn example can be found here: {}", example_config_file);
+                exit(1);
+            },
+            Ok(config) => config,
+        };
+        out
+    }
+}
+
+/// A function to make a configuration file.
+///
+/// # Arguments
+///
+/// * `config_file`: The location where the new config file should be placed.
+///
+/// returns: ()
+///
+/// # Examples
+///
+/// ```
+/// let config_file: String = String::from("config.toml");
+/// let future = make_default_config(&config_file);
+/// block_on(future);
+/// ```
+fn make_default_config(config_file: &String) {
+    // Try to create the file
+    let mut file = match File::create(config_file) {
+        Ok(f) => f,
+        Err(why) => panic!("Could not create the config file: {:?}.\n{}\n{}", why, PANIC_RESPONSE, why)
+    };
+
+    // Write the config file
+    match writeln!(&mut file, "# Config created automatically\n{}", include_str!("../config.toml.example")) {
+        Ok(_) => println!("Written default configuration to {}", config_file),
+        Err(why) => panic!("Could not write default configuration file to {}: {:?}\n{}\n{}", config_file, why, PANIC_RESPONSE, why)
+    };
+}
+
+/// Checks if the inputted config is compatible with the program
+///
+/// # Arguments
+///
+/// * `config_content`: The content of the config file to check for compatibility
+///
+/// returns: (bool, String)
+///
+/// # Examples
+///
+/// ```
+/// if ! check_version("config.toml")[0] {
+///     panic!("Config version incompatible!");
+/// }
+/// ```
+fn check_version(config_content: &String) -> (bool, String) {
+    // Get the version, by looping over the config contents, searching for the config line and
+    // getting the version part of it.
+    let version = {
+        let mut out = String::from("0.0");
+        let version_line = Regex::new("^version *= *\"\\d\\.\\d\" *(|#.*)$").unwrap();
+        let version_matcher = Regex::new("(\\d*\\.\\d*)").unwrap();
+        for line in config_content.lines() {
```

Comment:
This is not how you do regex!
I will do another PR later for that though.

Reply:
Does 1d73d5bf9172d34afd6a8f082411544279039e04 fix this issue for you?

Reply:
Not fully

Reply:
I will give you a patch when I'm finished

Reply:
Marking this as resolved for the other two comments